src/python/gem5/prebuilt/viper/board.py

```diff
@@ -0,0 +1,99 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import base64
+import os
+from typing import List
+
+from ...components.boards.kernel_disk_workload import KernelDiskWorkload
+from ...components.boards.x86_board import X86Board
+from ...components.cachehierarchies.abstract_cache_hierarchy import (
+    AbstractCacheHierarchy,
+)
+from ...components.devices.gpus.amdgpu import BaseViperGPU
+from ...components.memory.abstract_memory_system import AbstractMemorySystem
+from ...components.processors.abstract_processor import AbstractProcessor
+from ...utils.override import overrides
+
+
+class ViperBoard(X86Board):
+    """
+    A derivative of X86Board capable of full system simulation for X86 with a
+    GPU device. Provides all the functionality of the X86Board with helper
+    methods specific to booting a disk with GPU libraries installed.
+    """
+
+    def __init__(
+        self,
+        clk_freq: str,
+        processor: AbstractProcessor,
+        memory: AbstractMemorySystem,
+        cache_hierarchy: AbstractCacheHierarchy,
+    ) -> None:
+        super().__init__(
+            clk_freq=clk_freq,
+            processor=processor,
+            memory=memory,
+            cache_hierarchy=cache_hierarchy,
+        )
+
+    @overrides(KernelDiskWorkload)
+    def get_disk_device(self):
+        return "/dev/sda"
+
+    @overrides(KernelDiskWorkload)
+    def get_default_kernel_args(self) -> List[str]:
+        return [
+            "earlyprintk=ttyS0",
+            "console=ttyS0",
+            "lpj=7999923",
+            "root={root_value}",
+            "disk_device={disk_device}",
+            "drm_kms_helper.fbdev_emulation=0",
+            "modprobe.blacklist=amdgpu",
+            "modprobe.blacklist=psmouse",
+        ]
+
+    # Replicate the capability of the old GPUFS config, which embed a binary
+    # application or script into a bash script setting up the environment and
+    # loading the GPU driver.
+    def make_gpu_app(self, gpu: BaseViperGPU, app: str, debug: bool = False):
+        driver_load_command = gpu.get_driver_command(debug=debug)
+
+        with open(os.path.abspath(app), "rb") as binfile:
+            encodedBin = base64.b64encode(binfile.read()).decode()
+
+        application_command = (
+            f'echo "{encodedBin}" | base64 -d > myapp\n'
+            "chmod +x myapp\n"
+            "./myapp {}\n"
+            "/sbin/m5 exit\n"
+        )
+
+        return driver_load_command + application_command
```

Comment:
I'm not sure about line 87, but the rest of this should be instead in a `workload` object. We want to keep "workloads" (i.e., the thing you run on the board) separate from the "hardware" of the board.

Alternatively, you could add a *mixin* like `set_kernel_disk_workload` for the GPU applications. However, it is important that we keep it compatible with the `Workload` objects. Happy to provide more detail if needed.

Reply:
Should I then create something like GpuWorkload (in src/python/gem5/resources)?  There needs to be different driver load commands for different devices so that would be a base class for something like an MI300XWorkload, MI200Workload, etc.

Reply:
Yeah, I think that makes sense. It could also make sense to have a new mixin class. So, in addition to `set_kernel_disk_workload` we could also have a `set_gpufs_workload` if you need more parameters than what's available in `set_kernel_disk_workload`. It may be more efficient to discuss this at the dev meeting on Thursday.

I'm OK with merging this change without full support for using resources and workloads, and asking my team to implement that, but we need to make sure that we don't have to do a major refactoring.

Reply:
I don't fully understand the implications of this yet, but will need to.